pkg/tagger/collectors/kubelet_extract_test.go

```diff
@@ -8,57 +8,201 @@
 package collectors
 
 import (
-	"encoding/json"
-	"io/ioutil"
+	"fmt"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
-	"github.com/stretchr/testify/require"
 )
 
-func TestKubeletPodTags(t *testing.T) {
-	raw, err := ioutil.ReadFile("./testdata/kubelet/podlist_1.6.json")
```

Comment:
Not sure why you removed the json fixtures? seems more readable in a dedicated file

Reply:
Doing what I preach: using a dict-driven test allows faster maintainance as it's much easier to add a test case than modify the fixture .Although the initial setup is more cumbersome, it pays in the long term

Reply:
I'm agree with @xvello and I used this approach in `pkg/tagger/collectors/docker_extract_test.go`.

Reply:
I agree on dict-driven test 👍 but you won't test the json mapping anymore though

Reply:
the kubelet json to Pod objects logic is already tested in https://github.com/DataDog/datadog-agent/blob/master/pkg/util/kubernetes/kubelet/kubelet_test.go , and this package only uses `Pod` objects, so testing it here was redundant